src/components/grid.js

```diff
@@ -13,87 +13,145 @@ class Grid extends Component {
     constructor(props) {
         super(props);
         this.state = {
-            columnPosition: 0,
+            gridLayout: [],
+            rows: 0,
+            columns: 0,
             rowPosition: 0,
+            columnPosition: 0,
         };
-        this.columns = 0;
-        this.rows = 0;
-        this.buildGrid(this.props.gridSetup);
     }
 
     buildGrid(gridSetup) {
-        const gridLayout = [];
+        const currentGrid = [];
         for (let rowNum = 0; rowNum < gridSetup.length; rowNum++) {
-            gridLayout.push([]);
+            currentGrid.push([]);
 
-            for (let columnNum = 0; columnNum < gridSetup[rowNum].length; columnNum++) {
-                const squareItem = gridSetup[columnNum][rowNum];
+            for (
+                let columnNum = 0;
+                columnNum < gridSetup[rowNum].length;
+                columnNum++
+            ) {
+                const squareItem = gridSetup[rowNum][columnNum];
 
-                gridLayout[rowNum].push(squareItem);
+                currentGrid[rowNum].push(squareItem);
             }
         }
-        this.gridLayout = gridLayout;
-        this.rows = gridLayout.length;
-        this.columns = gridLayout[0].length;
-        console.log(gridLayout)
+
+        this.setState({
+            gridLayout: [...currentGrid],
+            rows: currentGrid.length,
+            columns: currentGrid[0].length,
+        });
     }
 
     componentWillMount() {
-        document.removeEventListener("keyup", this.onKeyPressed.bind(this));
+        document.removeEventListener('keyup', this.handleKeyPress.bind(this));
+        // document.querySelectorAll('.virtual-key').forEach((virtualKey) => {
+        //     virtualKey.removeEventListener(
+        //         'click',
+        //         this.handleKeyPress.bind(this)
+        //     );
+        // });
+
+        this.buildGrid(this.props.gridSetup);
     }
 
     componentDidMount() {
-		document.addEventListener("keyup", this.onKeyPressed.bind(this));
-	}    
-  
-	async onKeyPressed(e) {
-        e.stopPropagation()
-		switch(e.keyCode) {
-			case 38: {
-				// up
-                if (this.state.rowPosition > 0) {
-
-                    this.setState({ rowPosition: this.state.rowPosition - 1});
+        document.addEventListener('keyup', this.handleKeyPress.bind(this));
+        // document.querySelectorAll('.virtual-key').forEach((virtualKey) => {
+        //     virtualKey.addEventListener(
+        //         'click',
+        //         this.handleKeyPress.bind(this)
+        //     );
+        // });
+    }
+
+    checkSquareType(newRow, newColumn) {
```

Comment:
This method name is a little confusing - it should probably be more descriptive about what it actually does, since it moves the player. Just a nitpick though.